src/control/server/mgmt_system.go

```diff
@@ -874,7 +874,7 @@ func (svc *mgmtSvc) SystemCleanup(ctx context.Context, req *mgmtpb.SystemCleanup
 		return nil, errors.New("SystemCleanup requires a machine name.")
 	}
 
-	psList, err := svc.sysdb.PoolServiceList()
+	psList, err := svc.sysdb.PoolServiceList(true)
```

Comment:
Hmm. I'm not sure this is correct. This will give you a list of all pools in the system regardless of whether or not they're ready. If a pool is in the creating or destroying state, I don't think you want to clean up pool handles on it, right?